### PR TITLE
docs: website docs restructure

### PR DESCRIPTION
**Pull request by gyxlucy**
Add installation/self-hosted/cloud-gpu

### CODE REVIEW

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -184,6 +184,45 @@ const config = {
...
+            to: '/docs/installation/self-host/cloud-gpu/modal/',
+            from: '/docs/installation/modal/',
+          },
+          {
+            to: '/docs/installation/self-host/cloud-gpu/skypilot/',
+            from: '/docs/installation/skypilot/',
+          },
+          {
+            to: '/docs/installation/self-host/apple.md',
+            from: '/docs/installation/apple.md',
```
Comment (wsxiaoys) [inline website/docusaurus.config.js:205]:
```suggestion
            to: '/docs/installation/self-host/apple',
            from: '/docs/installation/apple',
```

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -184,6 +184,45 @@ const config = {
...
+          },
+          {
+            to: '/docs/installation/self-host/docker.mdx',
+            from: ['/docs/installation/docker.mdx', '/docs/installation/docker-compose.mdx'],
```
Comment (wsxiaoys) [inline website/docusaurus.config.js:209]:
```suggestion
            to: '/docs/installation/self-host/docker',
            from: ['/docs/installation/docker', '/docs/installation/docker-compose'],
```

File: website/docusaurus.config.js [hunk 3]
```diff
@@ -184,6 +184,45 @@ const config = {
...
+          },
+          {
+            to: '/docs/installation/model-hardware/models-registry.mdx',
+            from: '/docs/models/index.mdx',
```
Comment (wsxiaoys) [inline website/docusaurus.config.js:213]:
```suggestion
            to: '/docs/installation/model-hardware/models-registry',
            from: '/docs/models/',
```

File: website/docusaurus.config.js [hunk 4]
```diff
@@ -184,6 +184,45 @@ const config = {
...
+          },
+          {
+            to: '/docs/coding-assistant/ide/',
+            from: '/docs/extensions/',
+          },
+          {
+            to: '/docs/programming-languages.md',
+            from: '/docs/coding-assistant/ide/programming-languages.md',
```
Comment (wsxiaoys) [inline website/docusaurus.config.js:221]:
```suggestion
            to: '/docs/programming-languages',
            from: '/docs/coding-assistant/ide/programming-languages',
```

File: website/docs/installation/model-hardware/index.mdx
Comment (wsxiaoys) [inline website/docs/installation/model-hardware/index.mdx:1]: `model-hardware` -> `models`

**Comment by gyxlucy** [comment]
PTAL

File: website/docs/installation/models/models-leaderboard.mdx
Comment (wsxiaoys) [inline website/docs/installation/models/models-leaderboard.mdx:1]: `leaderboard.mdx`?

File: website/docs/installation/models/models-registry.mdx
Comment (wsxiaoys) [inline website/docs/installation/models/models-registry.mdx:1]: `registry.mdx`?

**Pull request merged**